libcaf_core/test/optional.cpp

```diff
@@ -4,10 +4,10 @@
 
 #define CAF_SUITE optional
```

Comment:
I think we can just remove this test suite entirely. I see no reason to keep it around.

Reply:
Doesn't it make sense to keep it around as long as `caf::optional` is not removed?